packages/techdocs-cli/src/commands/helpers.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ConfigReader } from "@backstage/config";
+import { Command } from "commander";
+
+export enum Publisher {
+  awsS3 = "awsS3",
+  googleGcs = "googleGcs",
+  azureBlobStorage = "azureBlobStorage",
+  openStackSwift = "openStackSwift",
+}
+
+type PublisherConfig = {
+  type: Publisher;
+  [Publisher.awsS3]?: any
+  [Publisher.googleGcs]?: any
+  [Publisher.azureBlobStorage]?: any
+  [Publisher.openStackSwift]?: any
+}
+
+/**
+ * Returns Backstage config suitable for use when instantiating a Publisher. If
+ * there are any missing or invalid options provided, an error is thrown.
+ */
+export const getValidPublisherConfig = (cmd: Command): ConfigReader => {
+  // Assuming that proper credentials are set in Environment variables
+  // for the respective GCS/AWS clients to work.
+  if (
+    !Object.keys(Publisher).includes(
+      cmd.publisherType
+    )
+  ) {
+    throw new Error(`Unknown publisher type ${cmd.publisherType}`);
+  }
+
+  let publisherConfig: PublisherConfig;
+  if (Publisher.azureBlobStorage === cmd.publisherType) {
```

Comment:
How about extracting functions in this same file or in a new file for each publisher and replacing those if-else with a switch statement?
It could make this file cleaner, leaving this function only with the responsibility of choosing an implementation for the config factory according to the `cmd.plubisherType`.

Reply:
Good idea.  I managed to get rid of the switch/case too using a typescript type guard and a class.  Let me know what you think!

Reply:
👏🏻 👏🏻 👏🏻